lib/node/nodes/dot-density.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var dot = require('dot');
+dot.templateSettings.strip = false;
+
+var Node = require('../node');
+
+var TYPE = 'dot_density';
+var PARAMS = {
+    source: Node.PARAM.NODE(Node.GEOMETRY.POLYGON),
+    points_col_name: Node.PARAM.NULLABLE(Node.PARAM.STRING),
```

Comment:
I would like to add @viddo to this conversation as this might affect a lot they way he is building the editor components to handle these optional params.

Other alternative is going more granular and have different analyses for different options:

1. For points
1. Points column + optional category column.